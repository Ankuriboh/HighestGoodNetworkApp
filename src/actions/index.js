export const getCurrentUser = token => {
	return {
		type: 'GET_CURRENT_USER',
		payload: token
	}
}
