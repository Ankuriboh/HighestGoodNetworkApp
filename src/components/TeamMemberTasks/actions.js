import { createAction } from "redux-actions";

export const fetchTeamMembersTaskBegin = createAction("FETCH_TEAM_MEMBERS_TASK_BEGIN");
export const fetchTeamMembersTaskSuccess = createAction("FETCH_TEAM_MEMBERS_TASK_SUCCESS");
export const fetchTeamMembersTaskError = createAction("FETCH_TEAM_MEMBERS_TASK_ERROR");