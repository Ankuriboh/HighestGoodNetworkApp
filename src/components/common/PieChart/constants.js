export const CHART_SIZE = 240;
export const CHART_RADIUS = CHART_SIZE / 2 - 20;

export const pieChartColors = ["rgb(255, 94, 130)", "rgb(179, 104, 210)", "rgb(100, 183, 255)", "rgb(255, 219, 86)", "rgb(94, 255, 219)", "rgb(255, 158, 180)", "rgb(255, 145, 69)", "rgb(0, 146, 178)"];
