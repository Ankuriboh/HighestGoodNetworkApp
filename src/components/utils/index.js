import moment from "moment";

export const getTimeEntries = week => {
  const { state } = this.props;
  let howManyDays;
  if (week === "Current Week") {
    howManyDays = 0;
  } else if (week === "Last Week") {
    howManyDays = 7;
  } else if (week === "Week Before Last") {
    howManyDays = 14;
  }

  const startWeek = moment()
    .subtract(howManyDays, "days")
    .startOf("week")
    .format("YYYY-MM-DD");
  const endWeek = moment()
    .subtract(howManyDays, "days")
    .endOf("week")
    .format("YYYY-MM-DD");

  getTimeEntryForSpecifiedPeriod(state.userProfile._id, startWeek, endWeek);
};

export const getTimeEntriess = week => {
  const { state } = this.props;
  let howManyDays;
  if (week === "Current Week") {
    howManyDays = 0;
  } else if (week === "Last Week") {
    howManyDays = 7;
  } else if (week === "Week Before Last") {
    howManyDays = 14;
  }

  const startWeek = moment()
    .subtract(howManyDays, "days")
    .startOf("week")
    .format("YYYY-MM-DD");
  const endWeek = moment()
    .subtract(howManyDays, "days")
    .endOf("week")
    .format("YYYY-MM-DD");

  getTimeEntryForSpecifiedPeriod(state.userProfile._id, startWeek, endWeek);
};
