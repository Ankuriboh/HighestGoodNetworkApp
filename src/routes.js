import React from 'react';
import {Route, IndexRoute } from 'react-router';
import App from './components/App';
import LoginPage from './components/Login/LoginPage';
import DashboardPage from './components/Dashboard/DashboardPage';
import UserManagementPage from './components/UserManagement/UserManagementPage';



export default (
    <Route path="/" component={App}>
        <Route path="Login" component={LoginPage} />
        <Route path="Dashboard" component={DashboardPage} />
        {/* <Route path="Timelog" component={TimelogPage}/>
        <Route path="Reports" component={ReportsPage}/>  */}
        <Route path="UserManagement" component={UserManagementPage}/> 
    </Route>
);