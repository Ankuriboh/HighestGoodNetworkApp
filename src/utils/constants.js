/**
 * Constant value to validate user permissions throughout the codebase
 */
export const permissions = {
  reports: 'getReports',
  weeklySummariesReport: 'getWeeklySummaries',
  userManagement: {
    getUserProfiles: 'getUserProfiles',
    postUserProfile: 'postUserProfile',
    deleteUserProfile: 'deleteUserProfile',
    changeUserStatus: 'changeUserStatus',
    putUserProfile: 'putUserProfile',
    addDeleteEditOwners: 'addDeleteEditOwners',
    updatePassword: 'updatePassword',
    manageTimeOffRequests: 'manageTimeOffRequests',
  },
  badgeManagement: {
    seeBadges: 'seeBadges',
    createBadges: 'createBadges',
    updateBadges: 'updateBadges',
    deleteBadges: 'deleteBadges',
  },
  projects: {
    postProject: 'postProject',
    deleteProject: 'deleteProject',
    putProject: 'putProject',
    getProjectMembers: 'getProjectMembers',
    assignProjectToUsers: 'assignProjectToUsers',
    unassignUserInProject: 'unassignUserInProject',
    postWbs: 'postWbs',
    deleteWbs: 'deleteWbs',
    postTask: 'postTask',
    updateTask: 'updateTask',
    deleteTask: 'deleteTask',
    suggestTask: 'suggestTask',
  },
  putUserProfileImportantInfo: 'putUserProfileImportantInfo',
  teams: {
    postTeam: 'postTeam',
    putTeam: 'putTeam',
    deleteTeam: 'deleteTeam',
    assignTeamToUsers: 'assignTeamToUsers',
  },
  permissionsManagement: {
    postRole: 'postRole',
    putRole: 'putRole',
    deleteRole: 'deleteRole',
    putUserProfilePermissions: 'putUserProfilePermissions',
  },
  inventoryProject: '',
  inventoryProjectWbs: '',
  popups: { createPopup: 'createPopup', updatePopup: 'updatePopup' },
  timeLog: {
    editTimelogInfo: 'editTimelogInfo',
    editTimeEntry: 'editTimeEntry',
    deleteTimeEntryOthers: 'deleteTimeEntryOthers',
    deleteTimeEntry: 'deleteTimeEntry',
  },
};
