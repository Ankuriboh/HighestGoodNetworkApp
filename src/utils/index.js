import moment from "moment";
import { store } from "../store";

export const getTimeEntries = (getTimeEntryForSpecifiedPeriod, id) => {
  const state = store.getState();
  let howManyDays;
  if (state.whichWeek === "Current Week") {
    howManyDays = 0;
  } else if (state.whichWeek === "Last Week") {
    howManyDays = 7;
  } else if (state.whichWeek === "Week Before Last") {
    howManyDays = 14;
  }
  const startWeek = moment()
    .subtract(howManyDays, "days")
    .startOf("week")
    .format("YYYY-MM-DD");
  const endWeek = moment()
    .subtract(howManyDays, "days")
    .endOf("week")
    .format("YYYY-MM-DD");

  return getTimeEntryForSpecifiedPeriod(id, startWeek, endWeek);
};

export const a = () => {
  console.log("a");
};
