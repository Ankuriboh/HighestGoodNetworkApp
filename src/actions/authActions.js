import jwtDecode from 'jwt-decode'
import httpService from '../services/httpService'
import config from '../config.json'
import { ENDPOINTS } from '../utils/URL'
import { GET_ERRORS } from '../constants/errors'
import { SET_CURRENT_USER } from '../constants/auth'

const { tokenKey } = config

export const loginUser = credentials => dispatch => {
  httpService
    .post(ENDPOINTS.LOGIN, credentials)
    .then(res => {
      if (res.data.new) {
        dispatch(setCurrentUser({ new: true, userId: res.data.userId }))
      } else {
        localStorage.setItem(tokenKey, res.data.token)
        httpService.setjwt(res.data.token)
        const decoded = jwtDecode(res.data.token)
        dispatch(setCurrentUser(decoded))
      }
    })
    .catch(err => {
      if (err.response && err.response.status === 403) {
        const errors = { email: err.response.data.message }
        dispatch({
          type: GET_ERRORS,
          payload: errors
        })
      }
    })
}

export const setCurrentUser = decoded => ({
  type: SET_CURRENT_USER,
  payload: decoded
})

export const logoutUser = () => dispatch => {
  localStorage.removeItem(tokenKey)
  httpService.setjwt(false)
  dispatch(setCurrentUser(null))
};
  