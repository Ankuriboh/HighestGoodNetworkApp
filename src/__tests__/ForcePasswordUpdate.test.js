import { renderWithProvider, renderWithRouterMatch } from './utils.js'
import '@testing-library/jest-dom/extend-expect'
import thunk from 'redux-thunk';
import mockState from './mockAdminState.js'
import { GET_ERRORS } from '../constants/errors';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { ENDPOINTS } from '../utils/URL';
import { render, fireEvent} from "@testing-library/react";
import routes from './../routes';
const url = ENDPOINTS.FORCE_PASSWORD;

const server = setupServer(
    rest.post(url, (req, res, ctx) =>  {
      console.log(req.body);
      // if (req.body.email === 'validEmail@gmail.com' && req.body.password === 'validPass') {
      //   return res(ctx.status(200), ctx.json({token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyaWQiOiI1ZWRmMTQxYzc4ZjEzODAwMTdiODI5YTYiLCJyb2xlIjoiQWRtaW5pc3RyYXRvciIsImV4cGlyeVRpbWVzdGFtcCI6IjIwMjAtMDgtMjhUMDU6MDA6NTguOTE0WiIsImlhdCI6MTU5NzcyNjg1OH0.zyPNn0laHv0iQONoIczZt1r5wNWlwSm286xDj-eYC4o'}), )
      // } else if (req.body.email === 'newUserEmail@gmail.com' && req.body.password === 'validPass'){
      //   return res(ctx.status(200), ctx.json({new: true, token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyaWQiOiI1ZWRmMTQxYzc4ZjEzODAwMTdiODI5YTYiLCJyb2xlIjoiQWRtaW5pc3RyYXRvciIsImV4cGlyeVRpbWVzdGFtcCI6IjIwMjAtMDgtMjhUMDU6MDA6NTguOTE0WiIsImlhdCI6MTU5NzcyNjg1OH0.zyPNn0laHv0iQONoIczZt1r5wNWlwSm286xDj-eYC4o'}), )
      // } else {
      //   return res(ctx.status(403), ctx.json({message: 'Invalid email and/ or password.'}), )
      // }
    }),  
    rest.get('http://localhost:4500/api/userprofile/*', (req, res, ctx) =>  {
        return res(ctx.status(200), ctx.json({}), )  
    }),
    rest.get('http://localhost:4500/api/dashboard/*', (req, res, ctx) =>  {
      return res(ctx.status(200), ctx.json({leaderBoardData: [
        {
          "personId": "5edf141c78f1380017b829a6",
          "name": "Dev Admin",
          "weeklyComittedHours": 10,
          "totaltime_hrs": 6,
          "totaltangibletime_hrs": 6,
          "totalintangibletime_hrs": 0,
          "percentagespentintangible": 100,
          "didMeetWeeklyCommitment": false,
          "weeklycommited": 10,
          "tangibletime": 6,
          "intangibletime": 0,
          "tangibletimewidth": 100,
          "intangibletimewidth": 0,
          "tangiblebarcolor": "orange",
          "totaltime": 6
        }]}), )  
    }),
    rest.get('*', (req, res, ctx) => {
      console.error(`Please add request handler for ${req.url.toString()} in your MSW server requests.`);
      return res(
        ctx.status(500),
        ctx.json({ error: 'You must add request handler.' }),
      );
    }),
  );
  
beforeAll(() => server.listen());
afterAll(() => server.close());
afterEach(() => server.resetHandlers());

import { loginUser } from "../actions/authActions";
import { describe } from 'joi';

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

describe('Force Password Update behaviour', () => {

  it('should update password after submit is clicked', async () => {
    // let rt = '/updatepassword/5edf141c78f1380017b829a6'
    // const hist = createMemoryHistory({ initialEntries: [rt] });
    // let loginMountedPage = renderWithRouterMatch(routes , {initialState: mockState, route: rt, history: hist});
    // await sleep(100);
    // let {getByLabelText, getByText} = loginMountedPage;
    // fireEvent.change(getByLabelText('Email:'), {
    //   target: {value: 'validEmail@gmail.com'}
    // });
    // await sleep(10);
    // fireEvent.change(getByLabelText('Password:'), {
    //   target: {value: 'validPass'}
    // });
    // await sleep(10);
    // fireEvent.click(getByText('Submit'));
    // await sleep(100);
    // expect(getByLabelText('Current Password:')).toBeTruthy();
    // return;
  });

});

describe('Force Password Update page structure', () => {
    it('should match the snapshot', () => {
      const { asFragment } = render(<ForcePasswordUpdate {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });