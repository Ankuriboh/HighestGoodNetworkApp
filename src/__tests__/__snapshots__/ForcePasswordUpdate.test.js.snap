// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Force Password Update page structure should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <h2>
      Change Password
    </h2>
    <form
      class="col-md-6 xs-12"
    >
      <div
        class="form-group "
      >
        <label
          for="newpassword"
        >
          New Password:
        </label>
        <div
          class="input-text"
        >
          <input
            class="form-control"
            id="newpassword"
            name="newpassword"
            type="password"
            value=""
          />
          <i
            class="fa fa-eye-slash"
          />
        </div>
      </div>
      <div
        class="form-group "
      >
        <label
          for="confirmnewpassword"
        >
          Confirm Password:
        </label>
        <div
          class="input-text"
        >
          <input
            class="form-control"
            data-refers="newpassword"
            id="confirmnewpassword"
            name="confirmnewpassword"
            type="password"
            value=""
          />
          <i
            class="fa fa-eye-slash"
          />
        </div>
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        style="box-shadow: 2px 2px 4px 1px lightgray;"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
